### PR TITLE
Value for query key

### PR DESCRIPTION
**Pull request by jdisho**
This PR 🚀 will add one extension to `URL`: value(for:) which returns the of a query key.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 1]
```diff
@@ -63,6 +63,20 @@ public extension URL {
 		self = appendingQueryParameters(parameters)
 	}
 
+    /// SwifterSwift: Get value of a query key.
+    ///
+    ///    var url = URL(string: "https://google.com?code=12345")!
+    ///    url.value(for: code) -> "12345"
+    ///
+    /// - Parameter: Key of a query.
+    public func value(for queryKey: String) -> String? {
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/Foundation/URLExtensions.swift]: Hey @jdisho I'm not sure about this naming here, to me `url.value(for:)` even with queryKey variable name,  doesn't make clear that is a value for a query parameter, seems like can be any value, I don't know, maybe something like `queryParameterValue(for key: String)`, but it has the con that is more verbose, but is more clear what it does.
What do you think? 
Note: I'm not good at naming this suggestion is probably not good naming at all 😂 
Let know your thoughts on that :))

File: Tests/FoundationTests/URLExtensionsTests.swift [hunk 2]
```diff
@@ -37,6 +37,12 @@ final class URLExtensionsTests: XCTestCase {
 		XCTAssertEqual(parameters["empty"], nil)
 	}
 
+    func testValueForQueryKey() {
```
Comment (LucianoPAlmeida) [inline Tests/FoundationTests/URLExtensionsTests.swift]: Also, can you add a test case for a non-existing key?

File: Sources/Extensions/Foundation/URLExtensions.swift (hunk 1 above)
Comment (SD10) [reply Sources/Extensions/Foundation/URLExtensions.swift]: I agree, and `value(for:)` is a pretty common API for stuff like NSUserDefaults and collections. Something like `queryValue` or `parameterValue` could be good names too. Good suggestion 👍

**Comment by jdisho** [comment]
Thanks for the feedback guys. I applied it!

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 3]
```diff
@@ -63,6 +63,20 @@ public extension URL {
 		self = appendingQueryParameters(parameters)
 	}
 
+    /// SwifterSwift: Get value of a query key.
+    ///
+    ///    var url = URL(string: "https://google.com?code=12345")!
+    ///    queryValue(for: "code") -> "12345"
+    ///
+    /// - Parameter: Key of a query.
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: This should be:
`/// - Parameter key: The key of a query value.`

Comment (jdisho) [reply Sources/Extensions/Foundation/URLExtensions.swift]: Good catch.

**Comment by SD10** [comment]
Awesome! Thanks @jdisho @LucianoPAlmeida @oliviabrown9 👍

**Pull request merged**